src/structures/GuildChannel.js

```diff
@@ -567,28 +560,31 @@ class GuildChannel extends Channel {
 
   /**
    * Whether the channel is deletable by the client user
-   * @type {boolean}
+   * @type {?boolean}
    * @readonly
    */
   get deletable() {
-    return this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false);
+    const permissions = this.permissionsFor(this.client.user);
+    return permissions && permissions.has(Permissions.FLAGS.MANAGE_CHANNELS);
```

Comment:
To be honest, I'm more of a fan of throwing an error here (as `permissionsFor` will only return `undefined` when somebody sweeps their member cache in a non-sensible way, aka not re-adding the client's member) as opposed to just returning `undefined`, but we can keep the current behaviour with:
```suggestion
    return permissions ? permissions.has(Permissions.FLAGS.MANAGE_CHANNELS) : false;
```

Same goes for many other places.

cc: @vladfrangu @iCrawl

Reply:
i think there should be some error handling here for when the permissions aren't returned, maybe a custom error like `UNCACHED_MEMBER` idk

Reply:
Sure, it's not an error everyone will see, only those who sweep caches.

Reply:
I agree with Kyra